data/preprocessing/nutrient_load_reduction/process_data.py

```diff
@@ -0,0 +1,92 @@
+""" Reads the limiting nutrients equal area vector file, reporjects the file to EPSG4326 and estimates the percentage of reduction needed to meet a good water quality conditions.
+
+Usage:
+process_data.py <folder>
+
+Arguments:
+    <folder>     Folder containing the limiting nutrients shapefile
+"""
+import os
+import logging
+from pathlib import Path
+import argparse
+
+import geopandas as gpd
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("preprocessing_limiting_nutrients_file")
+
+def check_and_reproject_to_4326(gdf):
+    """
+    Checks if a GeoDataFrame is in CRS 4326 (WGS84) and reprojects it if not.
+
+    Parameters:
+    - gdf: GeoDataFrame to check and reproject if needed.
+
+    Returns:
+    - Reprojected GeoDataFrame (if reprojected) or the original GeoDataFrame (if already in 4326).
+    """
+    if gdf.crs is None or gdf.crs.to_epsg() != 4326:
+        log.info("Reprojecting GeoDataFrame to EPSG:4326 (WGS84)...")
+        try:
+            # Reproject to EPSG:4326
+            gdf = gdf.to_crs(epsg=4326)
+            log.info("Reprojection successful.")
+        except:
+            log.error("Reprojection failed with error")
+    else:
+        log.info("GeoDataFrame is already in EPSG:4326 (WGS84).")
+
+    return gdf
+
+# Define the function to calculate perc_reduction
+def calculate_perc_reduction(row):
+    if row['Cases_v2_1'] == 4 and row['TP_con_V2_']:
+        return ((row['TP_con_V2_'] - 0.046) / row['TP_con_V2_']) * 100
+    elif row['Cases_v2_1'] == 2 and row['TN_con_V2_']:
```

Comment:
I have removed the comment above and added a more detailed description about that formula. Let me know what you think about it. We have to have in mind that the full description is already reflected on the LG methodology.

Reply:
it could done be in the function docstring rather than in a comment so the info and logic are part of the same structure :)
```suggestion
def calculate_perc_reduction(row):
    """Calculation of the required Load Reduction.
    
    This equation is applied to only the basin-specific limiting nutrient as identified by McDowell et al. (2020)
    The global concentration thresholds values for Total N (0.70 mg-N/L) and Total P (0.046 mg-P/L) represent
    acceptable levels of algal growth.
    
    More information can be found on the LandGriffon v2.0 methodology
    """
    if row['Cases_v2_1'] == 4 and row['TP_con_V2_']:
        return ((row['TP_con_V2_'] - 0.046) / row['TP_con_V2_']) * 100
    elif row['Cases_v2_1'] == 2 and row['TN_con_V2_']:
```